mongodb/activity/queryDocument/activity.go

```diff
@@ -0,0 +1,152 @@
+	"fmt"
+	"time"
+
+	"github.com/project-flogo/core/activity"
+	"github.com/project-flogo/core/data/coerce"
+	"github.com/project-flogo/core/data/metadata"
+	"github.com/project-flogo/core/support/log"
+	mongocon "github.com/project-flogo/datastore-contrib/mongodb/connection"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+var logquery = log.ChildLogger(log.RootLogger(), "mongodb-querydocument")
+
+func init() {
+	err := activity.Register(&Activity{}, New)
+	if err != nil {
+		fmt.Println(err)
+	}
+}
+
+// New functioncommon
+func New(ctx activity.InitContext) (activity.Activity, error) {
+	settings := &Settings{}
+	err := metadata.MapToStruct(ctx.Settings(), settings, true)
+	if err != nil {
+		return nil, err
+	}
+	if settings.Connection != "" {
+
+		mcon, toConnerr := coerce.ToConnection(settings.Connection)
+		if toConnerr != nil {
+			return nil, toConnerr
+		}
+		client := mcon.(*mongocon.MongodbSharedConfigManager).GetClient()
+		act := &Activity{client: client, operation: settings.Operation, collectionName: settings.CollectionName, database: mcon.(*mongocon.MongodbSharedConfigManager).GetClientConfiguration().Database}
+		return act, nil
+	}
+	return nil, nil
+}
+
+// Activity is a stub for your Activity implementation
+type Activity struct {
+	client         *mongo.Client
+	operation      string
+	collectionName string
+	database       string
+}
+
+var activityMd = activity.ToMetadata(&Input{}, &Output{})
+
+// Metadata implements activity.Activity.Metadata
+func (a *Activity) Metadata() *activity.Metadata {
+	return activityMd
+}
+
+//Cleanup method
+func (a *Activity) Cleanup() error {
+
+	logquery.Debugf("cleaning up MongoDB activity")
+
+	ctx, cancel := ctx.WithTimeout(ctx.Background(), 30*time.Second)
+	defer cancel()
+
+	return a.client.Disconnect(ctx)
+}
+
+// Eval implements activity.Activity.Eval
+func (a *Activity) Eval(context activity.Context) (done bool, err error) {
+	logquery.Debugf("Executing  mongodb query Activity")
+	method := a.operation
+	if method == "" {
+		return false, activity.NewError("operation is not configured", "MONGODB-1004", nil)
+	}
+	collectionName := a.collectionName
+	if collectionName == "" {
+		return false, activity.NewError("collection Name is not configured", "MONGODB-1004", nil)
+	}
+	var inputJSON string
+
+	inputVal := &Input{}
+	err = context.GetInputObject(inputVal)
+	if err != nil {
+		return true, nil
+	}
+
+	if inputVal.Input == nil {
+		// should we throw an error or warn?
+		logquery.Warnf("Nothing to insert")
+		return true, nil
+	}
+	jsonBytes, err := json.Marshal(inputVal.Input)
+	if err != nil {
+		return false, fmt.Errorf("Error reading input json %s", err.Error())
+	}
+	inputJSON = string(jsonBytes)
+	logquery.Debugf("InputJSON: ", inputJSON)
+	if err != nil {
+		return false, fmt.Errorf("Error getting mongodb connection %s", err.Error())
+	}
+
+	connectioInfo := a.database
+	client := a.client
+	db := client.Database(connectioInfo)
```

Comment:
why construct a temporary variable here?  why not use a.database directly?

Reply:
Yes, will remove